Home screen quick actions

eshurakov opened the pull request:
### Details
This PR adds home screen quick actions:
* Search: opens note list with active search
* New Note: opens new note
* Recent: opens recent note

@jleandroperez Another day another PR! Thank you!

Closes #950 

### Test
1. Long press on the app icon. 

- [x] Check that Search opens Note List and search bar is active
- [x] Check that New Note opens an empty editor 
- [x] Check that Recent opens a recently modified note

Please check when the app is in memory / killed.
Please also check when the app is in memory and some random screen is opened.

### Test (Log out)
1. Log out of the app
2. Long press on the app icon

- [x] Check that there are no quick actions available

### Release
`RELEASE-NOTES.txt` was updated in https://github.com/Automattic/simplenote-ios/pull/1032/commits/45dd28e13ef6325debded3730d9f5f841dc8b6f0 with:

> Home screen quick actions #950

eshurakov committed:
Add home screen quick actions

eshurakov committed:
Show recently modified note in home quick actions

eshurakov committed:
Switch from trash to all notes when executing search quick action

eshurakov committed:
Add search icon to home quick actions

eshurakov committed:
Update RELEASE-NOTES.txt

jleandroperez commented on Simplenote/SPAppDelegate+Extensions.swift:
```diff
@@ -69,6 +75,28 @@ extension SPAppDelegate {
+            // Switch from trash to all notes as trash doesn't have search
+            if self.selectedTag == kSimplenoteTrashKey {
+                self.selectedTag = nil
+            }
```
Since we're coming from the springboard, and the user is triggering Search, how about if we simply always ensure `All Notes` is selected?

eshurakov replied:
I checked this with Sly. It is fine to open the app wherever it was before. If tag was selected, we keep the tag selected. Search is global, so it won't be affected by the selected tag.
The only place that doesn't have search is Trash, that's why we switch it to All Notes.

jleandroperez commented on Simplenote/SPAppDelegate+Extensions.swift:
```diff
@@ -69,6 +75,28 @@ extension SPAppDelegate {
+            navigationController.popToViewController(noteListViewController, animated: false)
+        }
+
+        DispatchQueue.main.async {
```
Do we 100% need to do this async? (is it for animation purposes?)

eshurakov replied:
I did it similar to what we already had in app delegate. I will check if it can be done in a different way.

jleandroperez commented on Simplenote/SPAppDelegate.m:
```diff
@@ -412,18 +428,6 @@ - (NSPersistentStoreCoordinator *)persistentStoreCoordinator
-                                                      }
-                                                  }];
-    
-}
```
🖖

jleandroperez reviewed (changes requested):
Sending few nipticky notes sir!

1. **Shortcut:** Search feels a bit broken, when the editor was the last screen. Video sent!
2. **Shortcut:** New note doesn't have an animation
3. (Checking with Sly over GH) perhaps it's best if the Recent showed the last opened note. WIll confirm on this one!

Other than that, changes look super!

eshurakov committed:
Update note presentation from a shortcut

eshurakov committed:
Show recently opened note in quick actions

eshurakov commented:
Thank you for the review, @jleandroperez!

I've updated the code that handles quick actions, there should be less animations now and the user should just directly see the destination screen.

Also added quick actions cleanup when user logs out.

Ready for another look, thanks!

eshurakov committed:
Show passcode lock when opening from quick actions

jleandroperez commented on Simplenote/SPAppDelegate.m:
```diff
@@ -691,39 +686,6 @@ - (BOOL)application:(UIApplication *)app openURL:(NSURL *)url options:(NSDiction
-        [self.noteListViewController openNote:note animated:NO];
-        [self showPasscodeLockIfNecessary];
-    });
-}
```
Thank you for doing this!!

jleandroperez reviewed (approved):
Works super smoothly, @eshurakov thank you for fixing all the animation things!!

:shipit:
